src/app/data-client.ts

```diff
@@ -67,6 +69,65 @@ export class DataClient {
     return dataArray;
   }
 
+  async binaryDataByFilter(filter?: pb.Filter) {
+    const { service } = this;
+
+    let last = '';
+    const dataArray: pb.BinaryData.AsObject[] = [];
+
+    for (;;) {
+      const dataReq = new pb.DataRequest();
+      if (filter) {
+        dataReq.setFilter(filter);
+      } else {
+        dataReq.setFilter(new pb.Filter());
+      }
+      dataReq.setLimit(100);
+      dataReq.setLast(last);
```

Comment:
I know this is how the loop is set up in the `tabularDataByFilter` method, however I'm wonder if a new `DataRequest` needs to be created for each iteration. The filter and limit don't change between requests, just the `last` value. 

Could this work and slightly reduce the amount of work in each iteration?

```suggestion
    const dataReq = new pb.DataRequest();
    dataReq.setFilter(filter ?? new pb.Filter());
    dataReq.setLimit(100);
    
    for (;;) {
      dataReq.setLast(last);
```